skrf/calibration/deembedding.py

```diff
@@ -0,0 +1,452 @@
+
+    def __str__():
+        pass
+
+    def __repr_():
+        pass
+    
+    @abstractmethod
+    def deembed(self, ntwk):
+        '''
+        Apply de-embedding correction to a Network
+        '''
+        pass
+
+class OpenShort(Deembedding):
+    '''
+    Remove open parasitics followed by short parasitics. 
+    
+    This is a commonly used de-embedding method for on-wafer applications.
+
+    A deembedding object is created with two dummy measurements: `dummy_open` 
+    and `dummy_short`. When :func:`Deembedding.deembed` is applied, the 
+    Y-parameters of the dummy_open are subtracted from the DUT measurement, 
+    followed by subtraction of Z-parameters of dummy-short.
+
+    This method is applicable only when there is a-priori knowledge of the
+    equivalent circuit model of the parasitic network to be de-embedded,
+    where the series parasitics are closest to device-under-test, 
+    followed by parallel parasitics. For more information, see [1]_
+
+
+    Example
+    --------
+    >>> import skrf as rf
+    >>> from skrf.calibration import OpenShort
+
+    Create network objects for dummy structures and dut
+
+    >>> op = rf.Network('open_ckt.s2p')
+    >>> sh = rf.Network('short_ckt.s2p')
+    >>> dut = rf.Network('full_ckt.s2p')
+
+    Create de-embedding object
+
+    >>> dm = OpenShort(dummy_open = op, dummy_short = sh, name = 'test_openshort')
+        
+    Remove parasitics to get the actual device network
+        
+    >>> realdut = dm.deembed(dut)
+
+    References
+    ------------
+
+    .. [1] M. C. A. M. Koolen, J. A. M. Geelen and M. P. J. G. Versleijen, "An improved 
+        de-embedding technique for on-wafer high frequency characterization", 
+        IEEE 1991 Bipolar Circuits and Technology Meeting, pp. 188-191, Sep. 1991.
+    '''
+    def __init__(self, dummy_open, dummy_short, name=None, *args, **kwargs):
+        '''
+        Open-Short De-embedding Initializer
+
+        Parameters
+        -----------
+
+        dummy_open : :class:`~skrf.network.Network` object
+            Measurement of the dummy open structure
+
+        dummy_short : :class:`~skrf.network.Network` object
+            Measurement of the dummy short structure
+
+        name : string
+            Optional name of de-embedding object
+
+        args, kwargs:
+            Passed to :func:`Deembedding.__init__`
+
+        See Also
+        ---------
+        :func:`Deembedding.__init__`
+
+        '''
+        self.open = dummy_open.copy()
+        self.short = dummy_short.copy()
+        dummies = [self.open, self.short]
+
+        Deembedding.__init__(self, dummies, name, *args, **kwargs)
+
+    def deembed(self, ntwk):
+        '''
+        Perform the de-embedding calculation
+
+        Parameters
+        ----------
+        ntwk : :class:`~skrf.network.Network` object
+            Network data of device measurement from which
+            parasitics needs to be removed via de-embedding
+        '''

```

Comment:
Could you add in the docstring the Return section with the description of `caled`? Same for the other methods `deembed`